client/state/jetpack-connect/reducer.js

```diff
@@ -96,15 +101,49 @@ export function jetpackConnectAuthorize( state = {}, action ) {
 			return Object.assign( {}, state, { isAuthorizing: true, authorizeSuccess: false, authorizeError: false, autoAuthorize: true, userData: action.userData, bearerToken: action.data.bearer_token } );
 		case JETPACK_CONNECT_REDIRECT_WP_ADMIN:
 			return Object.assign( {}, state, { isRedirectingToWpAdmin: true } );
+		case JETPACK_CONNECT_SSO_AUTHORIZE:
+			return Object.assign( {}, defaultAuthorizeState, { autoAuthorize: true } );
+		case JETPACK_CONNECT_SSO_AUTHORIZATION_RECEIVE:
+			if ( action.error ) {
+				return Object.assign( {}, state, { autoAuthorize: false } );
+			}
 		case SERIALIZE:
 		case DESERIALIZE:
 			return Object.assign( {}, state, { isRedirectingToWpAdmin: false } );
```

Comment:
If you'd like to persist this state, I'd highly recommend adding a schema to avoid client errors in the future when this data structure changes. See: https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#solution-schema-validation